### PR TITLE
FEM-2677 companion ad support

### PR DESCRIPTION
**Pull request by giladna**
return new IMAConfig().setAdTagUrl(adTagUrl).
                                              setVideoMimeTypes(videoMimeTypes).
                                              setAlwaysStartWithPreroll(true).setAdLoadTimeOut(8).
                                              setCompanionAdConfig(companionAdSlot, 728, 90);

### CODE REVIEW

**Review by GouravSna — APPROVED** [review]

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/CompanionAdConfig.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.kaltura.playkit.plugins.ima;
+
+import android.view.ViewGroup;
+
+public class CompanionAdConfig {
+    private transient ViewGroup companionAdView;
+    private int companionAdWidth = 728;
+    private int companionAdHeight = 90;
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/CompanionAdConfig.java]: Remove the defaults

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAConfig.java [hunk 2]
```diff
@@ -267,23 +278,14 @@ public IMAConfig addControlsOverlay(View controlsOverlay) {
...
     }
 
-    //    public Map<Double, String> getTagsTimes() {
+//    public Map<Double, String> getTagsTimes() {
 //        return tagsTimes;
 //    }
 //
 //    public void setTagsTimes(Map<Double, String> tagsTimes) {
 //        this.tagsTimes = tagsTimes;
 //    }
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAConfig.java]: Remove

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java [hunk 3]
```diff
@@ -279,29 +282,54 @@ protected void onUpdateMedia(PKMediaConfig mediaConfig) {
...
+        }
+
+
+        adCompanionViewGroup = (adConfig.getCompanionAdConfig() != null && adConfig.getCompanionAdConfig().getCompanionAdView() != null) ? adConfig.getCompanionAdConfig().getCompanionAdView() : null;
+
         if (adDisplayContainer != null) {
             log.d("adDisplayContainer != null return current adDisplayContainer");
             adDisplayContainer.unregisterAllVideoControlsOverlays();
             registerControlsOverlays();
+            if (adConfig.getCompanionAdConfig() != null && adCompanionViewGroup != null) {
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java]: Check width, height too

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java [hunk 4]
```diff
@@ -963,6 +992,9 @@ public void onAdError(AdErrorEvent adErrorEvent) {
             messageBus.post(new AdEvent.AdRequestedEvent(!TextUtils.isEmpty(adConfig.getAdTagUrl()) ? adConfig.getAdTagUrl() : adConfig.getAdTagResponse()));
         }
         sendError(errorType, errorMessage, adException);
+        if (PKAdErrorType.COMPANION_AD_LOADING_FAILED.equals(errorType)) {
```
Comment (noamtamim) [inline imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java:1006]: Check if we can report non-fatal error

**Review by noamtamim — CHANGES_REQUESTED** [review]

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/CompanionAdConfig.java (hunk 1 above)
Comment (giladna) [reply imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/CompanionAdConfig.java]: done

File: imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java (hunk 4 above)
Comment (giladna) [reply imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java:1006]: done

**Review by noamtamim — APPROVED** [review]

**Pull request merged**